### PR TITLE
Making template more generic and descriptive

### PR DESCRIPTION
**Pull request by xanmanning**
## Description

Changes to documentation explaining the purpose of each section as well as attempting to make the template more Koodoo/Kudu specific. (Original was based off an early draft from my previous employer 🙈 )

### Summary

  - Description tweaked to say include a Jira ticket
  - Summary updated to prompt author to ensure points in this section bring important changes to the forefront of reviewer's mind
  - Prerequisites purpose refined, ie. Do I need extra tooling, config, accounts, etc.
  - Commitment to review attempts to highlight the potential value of this section to your team mates.
  - Review instructions prompt you to include links where possible, trying to make this as frictionless to co-workers who don't work in the same discipline as you.
  - Acceptance Criteria, updated description - removed references to Travis-CI as we don't use this.
  - Added a references section, this is optional but can be valuable for allowing reviewers to understand decisions made by the author.

### Commitment to review

✋  This may trigger a lot of thought, opinions and debate, so could take a while to review. Very low priority, so don't prioritise it over sprint work.

  - Reviewing changes takes about _15-60 minutes_. (depends on how passionate you feel about this)
  - Reviewing _1 files_.

### Review instructions

  - Read through the comments [of the raw markdown](https://raw.githubusercontent.com/KoodooHQ/.github/fix/make-more-generic/PULL_REQUEST_TEMPLATE.md) (HINT: try reading in VSCode). I'm looking for comments or potential improvements regarding
    - Purpose of each section
    - Level of descriptiveness of each section (too much, too little?) - keep in mind that after the first 3-4 reviews, familiarity with the template will mean PRs descriptions will be quicker to write
    - Value of each section
    - Are there additional sections that you feel might help you review code?
    - Is this generic enough for your use case?
    - Spelling/grammar

### Acceptance criteria

  - [ ] Purpose of each section is clear
  - [ ] Each section has an adequate level of descriptiveness
  - [ ] Each section has some value
  - [ ] Template is fairly generic and can be used across a wide selection of code projects
  - [ ] Spelling and grammar is correct
  - [ ] There's a reasonable level of agreement from a variety of teams that the template changes are acceptable
  - [ ] LGTM :+1:

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -2,32 +2,48 @@
...
-  - `molecule < 3.0.0`
-  - `terraform > 0.12`
+Do I need any specific tools, applications, account access or configuration
+to review and manually test this PR? Below are some examples, if specific
+versions required, ensure this is specified:
+
+  - `jq`
+  - `docker`
+  - `yarn`
+  - `poetry`
```
Comment (sra405) [inline PULL_REQUEST_TEMPLATE.md:26]: 🥳

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -2,32 +2,48 @@
...
+  - `terraform >= 0.12`
```
Comment (sra405) [inline PULL_REQUEST_TEMPLATE.md]: soz grandad
```suggestion
  - `terraform >= 1.0.0`
```

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -56,24 +83,36 @@ For example:
...
+code quality or style, if you feel it is relevant or would like your reviewer
+to consider this include this in the checklist.
 
 Examples:
 
   - [ ] AC1: Molecule tests pass
   - [ ] AC2: NTP config is updated as expected
   - [ ] AC3: README has been updated to include the new NTP options
-  - [ ] Tests pass in Travis-CI
+  - [ ] Tests pass in GitHub Actions
```
Comment (sra405) [inline PULL_REQUEST_TEMPLATE.md]: we've not eradicated it yet 😢 
```suggestion
  - [ ] Tests pass in GitHub Actions/CircleCI
```

**Review by sra405 — APPROVED** [review]
🙌

**Review by scslingsby — APPROVED** [review]
👍   LGTM although I suppose my personal preference would be to have no comment sections at all `<!-- comments -->`, but not really a big deal.

**Review by jose-verissimo — APPROVED** [review]
LGTM, I use this template all of the time so looking forward to not having to copy it every damn time 👍

**Review by peelyKoodoo — APPROVED** [review]

**Pull request merged**